### ISSUE TITLE
[Bug+FR]: Source repo for unlisted plugin does not change if removed and reinstalled from community plugins

### ISSUE THREAD

**Issue by anonhostpi** [4Source/vare-obsidian-plugin#11]
### Describe the bug

I recently forked the teleprompter plugin, and published my own release files for it. To add it into Obsidian I removed the original and added mine as an "unlisted" plugin. After getting my changes implemented on the upstream repo, I removed the unlisted plugin and reinstalled the official one.

However, VARE is still pulling releases from my repo despite reinstalling the plugin.

I recommend setting an option _to change/update (and reset if applicable) the source repo in VARE's options_

I also recommend showing a hint text that shows the currently selected source repository in addition to these:

![image](https://github.com/4Source/vare-obsidian-plugin/assets/130698314/055846d0-484b-437c-8b9c-fd1e9105437b)

### How to Reproduce

Fork a plugin. Add the fork as an unlisted repo. Uninstall the plugin. Reinstall the official plugin.

### Expected behavior

Update the source repo on reinstallation

### OS of your device

Win10

### Obsidian Version

v1.6.3

### Plugin Version

0.1.1

### Additional context

This is a dual bug report + feature request.

I'd like to be able to change the source repo as well.

**Comment by anonhostpi** [comment]
This is also a feature request to get a button or text field to update/change the source repo

**Comment by anonhostpi** [comment]
Current work around to reset the repo.

Change the repo value in .obsidian\plugins\vare\data.json, save, then reload obsidian

**Comment by 4Source** [comment]
BUG: 
I will check the Bug but you need to wait about a month then I finished my exams and I have time to work on this. 

FEAT:
I think about add an edit button to the plugins BUT I mean I had this thoughts while dev and decided not to include it because any repo could be added and this could result in difficult to resolve problems. Example:
Other Plugin URL is entered and a version for this is selected the plugin get installed to the old folder but from the manifest it gets a new name. Than the folder name and plugin name doesn't match. 

> I recently forked the teleprompter plugin, and published my own release files for it. To add it into Obsidian I removed the original and added mine as an "unlisted" plugin.

I had roughly the same idea that you can also install releases from forks. See #9. Maybe this is anything you need?

**Comment by anonhostpi** [comment]
You use app.plugins.installPlugin() under the hood to install the plugin. That method installs the plugin to `.obsidian/plugins/{manifest.id}`, so at installation time, it is impossible for the folder and the plugin id to be 2 different values.

_The problematic behavior that this does cause_ is that if the manifest.id does not match the plugin listed in VARE, it will install the plugin as _an additional plugin_ or _overwrite another plugin_.

To implement the edit button safely, I would enforce that the upstream manifest.id must be equal to the manifest.id of the associate plugin. If the ids mismatch, throw an error message and tell the user to use the add (+) button instead.

With that said, I would specifically enforce that the edit and add buttons be separate:

- If the _add_ button trys to pull a plugin with an id that's already installed, it should throw an error and tell the user to use the _edit_ button
- If the _edit_ button trys to pull an update that has a mismatched id, it should throw an error tell the user to use the _add_ button

**Comment by anonhostpi** [comment]
The manifest.dir that you use in your plugin is populated and overwritten at runtime by app.plugins.loadManifests() and app.plugins.loadManifest()

**Comment by anonhostpi** [comment]
Also your plugin mentions the ability to install themes, but I don't see app.customCss.installTheme( manifest, legacyBool ) used anywhere

**Comment by anonhostpi** [comment]
Also how are themes intended to be installed? I don't see any calls to app.customCss.installLegacyTheme( manifest ) or app.customCss.installTheme( manifest, true /*set to true so that manifest.repo can be used for install*/)?

**Comment by 4Source** [comment]
> The manifest.dir that you use in your plugin is populated and overwritten at runtime by app.plugins.loadManifests() and app.plugins.loadManifest()

What do you mean? [This](https://github.com/4Source/vare-obsidian-plugin/blob/307ba0e49a2a89ca42225e6165835039b020a050/src/settings/SettingsTab.ts#L195)?



> Also how are themes intended to be installed? I don't see any calls to `app.customCss.installLegacyTheme( manifest )` or `app.customCss.installTheme( manifest, true /*set to true so that manifest.repo can be used for install*/)`?

Templates is currently not implemented

**Comment by anonhostpi** [comment]
Yup. When Obsidian loads the manifests for either apps.customCss or apps.plugins, it overwrites that JSON property

**Comment by 4Source** [comment]
It's been a while but with the brief look it seems like I exactly did this
